pkg/crawl/crawler.go

```diff
@@ -51,8 +53,14 @@ type ProcessFunc func(ctx context.Context, path string, d fs.DirEntry, result Fi
 // the path should be processed by the provided process function. On encountering a directory, the path will be compared
 // against all IgnoreDirs configured in the Crawler. If any pattern matches, the directory (and all files nested inside
 // the directory) will be ignored.
-func (c Crawler) Crawl(ctx context.Context, root string, match MatchFunc, process ProcessFunc) (Stats, error) {
+func (c Crawler) Crawl(ctx context.Context, root string, directoryCrawlRateLimit int, match MatchFunc, process ProcessFunc) (Stats, error) {
```

Comment:
Not particularly opinionated on this but wonder if the limit value might be better added to the `Crawler` struct rather than passed into the function, that way in tests for the default value of `0` we can carry on just using `Crawler{}`

Reply:
Yes, I think you're right, will swap

Reply:
Okay, I've pushed a compromise